grc/core/blocks/embedded_python.py

```diff
@@ -87,7 +87,6 @@ def __init__(self, flow_graph, **kwargs):
         self.states['_io_cache'] = ''
 
         self.module_name = self.name
-        self._epy_source_hash = -1
```

Comment:
I see you're removing the hash entirely, are you planning on bringing it back in a later PR?

Reply:
No; unless you have a specific use for it in mind that I'm missing :). 

(As far as I can tell) before the change to the IO extraction, the source hash mechanism was strictly an optimization to avoid a redundant block instantiation, extraction and port update; without affecting any observable behavior (though I don't believe I have tested this in isolation...). The only use of the `self._epy_source_hash` variable in the source code was to return early from the `rewrite()` function if the hash of the embedded python source code had not change. Since I had to remove this early return in order to allow for the second (final) instantiation/extraction of the python block with the final parameter values the source hash and the code calculating it were now dead code, thus I removed them. (see efa9a3f955cd87333ce926f41c5e9f2c95c40762)

*Pedantic side note:* I believe that the hash check in itself for the purpose of detecting a **lack of change** was technically invalid in the first place: given that the hash function used for Python strings isn't a cryptographic hash function it would typically only be used to quickly **distinguish** objects (matching hashes would result in a full comparison in a hash table). Admittedly it would have been very unlikely to get a hash collision by accident, but that would have been quite the Heisenbug :smiley:.